New Component DatePickerFullScreen

cerinman opened the pull request:
@mxenabled/frontend-engineers 

This PR adds a new component.  DatePickerFullScreen.

This component is a copy of the DatePicker component but the calendar is now a full screen scrim that covers it's parent element.

![screen shot 2015-11-30 at 3 41 59 pm](https://cloud.githubusercontent.com/assets/6463914/11486966/882e54c0-9779-11e5-97e3-a2c0f4180074.png)
![screen shot 2015-11-30 at 3 42 16 pm](https://cloud.githubusercontent.com/assets/6463914/11486965/882c02a6-9779-11e5-96f9-53f000118644.png)


Props:

- isFixed Type(boolean) - Default(False): Used to determine if the calendar scrim is positioned fixed or absolute.
- caretWrapperStyle Type(object): Used to style the caret icon on the selected date.
- closeIcon Type(string) - Default('close'): Name of icon used to select the close icon on the calendar scrim.
- closeOnDateSelect Type(boolean) - Default(False): Determines if the calendar scrim is closed on date select or not.
- defaultDate Type(number): Unix time stamp. The default date for the calendar and selected date.
- format Type(string) - Default('MMM D, YYYY'): The format for the selected date.
- inputStyle Type(object): Used to style the selected date input.
- locale Type(string) - Default('en'): Used for time zone offset when formatting dates.
- minimumDate Type(number): Unix time stamp. Sets the earliest date which a user can select.
- onDateSelect Type(function): A supplied call back function that fires on date select and returns the selected date.
- placeholderText Type(string): The place holder text for the selected date input field.
- placeholderTextStyle Type(object): Used to style the place holder text.
- selectedDateWrapperStyle Type(object): Used to style the selected date div wrapping element.
- showCaret Type(boolean) - Default(true): Determines if the caret icon is shown on the selected date.
- showDayBorders Type(boolean) - Default(false): Determines if the calendar dates have borders or not.
- style Type(object): Used to style component.
- title Type(string) - Default('Select A Date'): The title placed above the calendar.
- useInputForSelectedDate Type(boolean) - Default(true): Determines if the selected date is rendered as plain text or placed in an input field.

Classes:

- mx-date-picker-full-screen: Used to style component.
- mx-date-picker-full-screen-selected-date: Used to style the selected date.
- mx-date-picker-full-screen-calendar-scrim: Used to style the calendar scrim.
- mx-date-picker-full-screen-calendar-wrapper: Used to style the div wrapping the calendar.
- mx-date-picker-full-screen-calendar-header: Used to style the calendar header.

bsbeeks commented on src/components/DatePickerFullScreen.js:
```diff
@@ -0,0 +1,518 @@
+}
+
+DatePickerFullScreen.propTypes = {
+  calendarScrimFixedPosition: React.PropTypes.bool,
```
I would call this `isFixed` instead.

bsbeeks commented on src/components/DatePickerFullScreen.js:
```diff
@@ -0,0 +1,518 @@
+
+const styles = {
+  caret: {
+    color: '#CCCCCC',
```
Could we update all the colors to be pulled from the Style constants file? Refer to frontify for the color pallet to use.

cerinman commented:
@bsbeeks fixed naming issue and moved all font sizes/colors to values from the constants file.  Also removed px where ever appropriate.  

Here is is after the changes.
![screen shot 2015-12-01 at 8 56 21 am](https://cloud.githubusercontent.com/assets/6463914/11505841/9c205be6-9809-11e5-9b81-0fa98fe6e6f1.png)
![screen shot 2015-12-01 at 8 56 29 am](https://cloud.githubusercontent.com/assets/6463914/11505842/9c2924a6-9809-11e5-9c1d-08c0f33b559a.png)

bsbeeks commented:
One last change, we need to be able to change the close icon through a prop.

cerinman commented:
@bsbeeks The prop has been added. Ready for another review.

bsbeeks commented:
:shipit: make sure we update the docs once we release this.
